apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -191,6 +196,45 @@ export type Lti13ClaimType = z.infer<typeof Lti13ClaimSchema>;
 
 export const STUDENT_ROLE = 'http://purl.imsglobal.org/vocab/lis/v2/membership#Learner';
 
+export async function getOpenidClientConfig(
+  lti13_instance: Lti13Instance,
+  options?: client.ModifyAssertionOptions,
+): Promise<client.Configuration> {
+  const keystore = lti13_instance.keystore as jose.JSONWebKeySet | null | undefined;
+  if (!keystore || !Array.isArray(keystore.keys) || keystore.keys.length === 0) {
+    throw new HttpStatusError(403, 'LTI 1.3 configuration error: no keys available in keystore');
+  }
+
+  const keyFromKeyStore = keystore.keys.at(-1);
+  if (!keyFromKeyStore) {
+    throw new HttpStatusError(403, 'LTI 1.3 configuration error: unable to load key');
+  }
+
+  const privateKey: client.PrivateKey = {
+    // Provide alg explicitly to avoid import failures when JWKs omit `alg`
+    key: (await jose.importJWK(
+      keyFromKeyStore,
+      keyFromKeyStore.alg ?? 'RS256',
+    )) as client.CryptoKey,
```

Comment:
Rather than using `as ...`, it's usually better to make runtime assertions that narrow the type. In this case, asserting that it's not a `Uint8Array` or that a property like `algorithm` exists would be better.

Reply:
Addressed in fa49a1896f